src/components/Hoverable/ActiveHoverable.tsx

```diff
@@ -0,0 +1,137 @@
+import {cloneElement, forwardRef, Ref, useCallback, useEffect, useImperativeHandle, useMemo, useRef, useState} from 'react';
+import {DeviceEventEmitter} from 'react-native';
+import assignRef from '@libs/assignRef';
+import CONST from '@src/CONST';
+import HoverableProps from './types';
+
+type ActiveHoverableProps = Omit<HoverableProps, 'disabled'>;
+
+function ActiveHoverable({onHoverIn, onHoverOut, shouldHandleScroll, children}: ActiveHoverableProps, outerRef: Ref<HTMLElement>) {
+    const [isHovered, setIsHovered] = useState(false);
+
+    const elementRef = useRef<HTMLElement | null>(null);
+    const isScrollingRef = useRef(false);
+    const isHoveredRef = useRef(false);
+
+    const updateIsHovered = useCallback(
+        (hovered: boolean) => {
+            isHoveredRef.current = hovered;
+            if (shouldHandleScroll && isScrollingRef.current) {
+                return;
+            }
+            setIsHovered(hovered);
+        },
+        [shouldHandleScroll],
+    );
+
+    // Expose inner ref to parent through outerRef. This enable us to use ref both in parent and child.
+    useImperativeHandle<HTMLElement | null, HTMLElement | null>(outerRef, () => elementRef.current, []);
```

Comment:
Why `useImperativeHandle` instead of `forwardRef`?

Reply:
If `BoundsObserver` is the single reason we have to do the "tricky stuff", we can consider changing the `BoundsObserver` implementation. But I don't have any specific idea.

The relevant line in `BoundsObserver` is probably [this one](https://github.com/react-ng/bounds-observer/blob/2ff5fff1aedb8451c01829c84757ebcdb97edc14/src/index.tsx#L121).

Reply:
> Requires an HTML element ref from the direct children 

Yes, it requires its direct child to be "compatible" with [this method of retrieving the HTML element](https://github.com/react-ng/bounds-observer/blob/2ff5fff1aedb8451c01829c84757ebcdb97edc14/src/index.tsx#L36).